core/test/qa_DynamicBlock.cpp

```diff
@@ -29,54 +30,84 @@ static_assert(gr::traits::block::input_ports<fixed_source<int>>::size() == 0);
 static_assert(gr::traits::block::output_ports<fixed_source<int>>::size() == 1);
 
 template<typename T>
-struct cout_sink : public gr::Block<cout_sink<T>, gr::PortInNamed<T, "in">> {
-    std::size_t remaining = 0;
+struct debug_sink : public gr::Block<debug_sink<T>, gr::PortInNamed<T, "in">> {
```

Comment:
Minor: 
* naming convention -> `debug_sink` -> `DebugSink`
* Boy-scout rule: 
    * check if there isn't already an existing sink that does the same (remove code duplication)
    * port should be a field member + refl-annotated <-> this is the user-code reference that we want to follow. The tuple-defined ports should be reserved primarily for meta-programming/non-user defined blocks since those C++ concepts are a bit harder to grasp for C++ beginners.

Reply:
> check if there isn't already an existing sink that does the same (remove code duplication)

I checked and didn't find any. But it might still be worthwhile to move `DebugSink` from `qa_DynamicBlock` to `common_blocks.hpp`, as this kind of functionality is nice to have for automated tests. Maybe also rename it to something more suitable (`TestSink` ?) and generalize its interface so that it can debug/test multiple input values (not just one).

Reply:
Agreed ... there are several multiple copies of these type of testing/debugging sinks that should be streamlined.

Reply:
Maybe we could have a general `InspectBlock`, which acts as a pass-through block for debugging purposes that holds all the values coming in (so that they can be used in tests) and then pushes them to `n` output ports unaltered. If `n` is `0`, then it is like the sink described above.